doc/porting/arch.rst

```diff
@@ -458,3 +458,104 @@ Each architecture also needs its own linker script, even if most sections can
 be derived from the linker scripts of other architectures. Some sections might
 be specific to the new architecture, for example the SCB section on ARM and the
 IDT section on x86.
+
+Hardware Stack Protection
+=========================
+
+This option uses hardware features to generate a fatal error if a thread
+in supervisor mode overflows its stack. This is useful for debugging, although
+for a couple reasons you can't reliably make any assertions about the state
+of the system after this happens:
+
+* The kernel could have been inside a critical section when the overflow
+  occurs, leaving important global data structures in a corrupted state.
+* For systems that implement stack protection using a guard memory region,
+  it's possible to overshoot the guard and corrupt adjacent data structures
+  before the hardware detects this situation.
+
+To enable the :option:`CONFIG_HW_STACK_PROTECTION` feature, the system must
+enable some kind of hardware-based stack overflow protection, and enable the
+:option:`CONFIG_ARCH_HAS_STACK_PROTECTION` option.
+
+There are no C APIs that need to be implemented to support this, this is
```

Comment:
How about:

     There are no C APIs that need to be implemented to support stack protection,
     and it's entirely implemented within the ``arch/`` code.